### DISCUSSION TITLE
Proper waiting for file to download

### DISCUSSION THREAD

**Discussion by Kubas556** [dokan-dev/dokan-dotnet#362] [Q&A] [answered]
Hi, I would like to ask how to properly handle read operation, when the file is still downloading. I searched the microsoft documentation of NtReadFile function and found nothing about this case. Closest is to return STATUS_PENDING status, but after that the read function will never be called again. 

Thanks.

**Comment by Liryna** [comment]
Hi @Kubas556 ,
`STATUS_PENDING` shouldn't be returned by userland filesystem. Dokany actually already respond to IRP with `STATUS_PENDING` to give you time to process the read. What you need to do is to wait for the download to happen, maybe call [TryResetTimeout](https://github.com/dokan-dev/dokan-dotnet/blob/fa5b28d7377e47a3631a3832c2ce80a36d9fc89f/DokanNet/IDokanFileInfo.cs#L67-L72) if it takes too long and return the result when you can.

**Reply by Kubas556** [reply, marked as answer]
I will try that
